packages/grpc-js/src/channelz.ts

```diff
@@ -182,105 +199,64 @@ export class ChannelzTrace {
 }
 
 export class ChannelzChildrenTracker {
-  private channelChildren: Map<number, { ref: ChannelRef; count: number }> =
-    new Map<number, { ref: ChannelRef; count: number }>();
-  private subchannelChildren: Map<
+  private channelChildren = new OrderedMap<
+    number,
+    { ref: ChannelRef; count: number }
+  >();
+  private subchannelChildren = new OrderedMap<
     number,
     { ref: SubchannelRef; count: number }
-  > = new Map<number, { ref: SubchannelRef; count: number }>();
-  private socketChildren: Map<number, { ref: SocketRef; count: number }> =
-    new Map<number, { ref: SocketRef; count: number }>();
+  >();
+  private socketChildren = new OrderedMap<
+    number,
+    { ref: SocketRef; count: number }
+  >();
+  private trackerMap = {
+    [EntityTypes.channel]: this.channelChildren,
+    [EntityTypes.subchannel]: this.subchannelChildren,
+    [EntityTypes.socket]: this.socketChildren,
+  } as const;
 
   refChild(child: ChannelRef | SubchannelRef | SocketRef) {
-    switch (child.kind) {
-      case 'channel': {
-        const trackedChild = this.channelChildren.get(child.id) ?? {
-          ref: child,
-          count: 0,
-        };
-        trackedChild.count += 1;
-        this.channelChildren.set(child.id, trackedChild);
-        break;
-      }
-      case 'subchannel': {
-        const trackedChild = this.subchannelChildren.get(child.id) ?? {
-          ref: child,
-          count: 0,
-        };
-        trackedChild.count += 1;
-        this.subchannelChildren.set(child.id, trackedChild);
-        break;
-      }
-      case 'socket': {
-        const trackedChild = this.socketChildren.get(child.id) ?? {
-          ref: child,
-          count: 0,
-        };
-        trackedChild.count += 1;
-        this.socketChildren.set(child.id, trackedChild);
-        break;
-      }
+    const tracker = this.trackerMap[child.kind];
+    const trackedChild = tracker.getElementByKey(child.id);
+
+    if (trackedChild === undefined) {
+      tracker.setElement(child.id, {
+        // @ts-expect-error union issues
```

Comment:
just typescript being silly and not generalizing the type good enough, I'll see if I can work around that

Reply:
adjusted this a little, but its give or take the same, just not an expect error, so it will be more noticeable if there is something other than the typing bug